src/hook/useCustomForm.js

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from "react";
+
+const useCustomForm = (initialInput, validate) => {
+  const [input, setInput] = useState(initialInput);
+  const [errors, setErrors] = useState({});
+
+  const validateInput = () => {
+    const validationErrors = validate(input);
+    setErrors(validationErrors);
+    return Object.keys(validationErrors).length === 0;
+  };
+
+  const register = (e) => {
```

Comment:
rules will be pass in the register with validations

Reply:
Done